packages/core/src/lib/history.integration.test.ts

```diff
@@ -0,0 +1,94 @@
+import { mkdir, rm, writeFile } from 'node:fs/promises';
+import { join } from 'node:path';
+import { type SimpleGit, simpleGit } from 'simple-git';
+import { afterAll, beforeAll, describe, expect } from 'vitest';
+import { getHashes } from './history';
+
+// we need a separate folder that is not cleaned in `global-setup.ts`, otherwise the tests can't execute in parallel
+const gitTestFolder = 'core-history-git-test';
+describe('git utils in a git repo', () => {
+  const baseDir = join(process.cwd(), gitTestFolder);
+  let emptyGit: SimpleGit;
+
+  beforeAll(async () => {
+    await mkdir(baseDir, { recursive: true });
+    emptyGit = simpleGit(baseDir);
+    await emptyGit.init();
+    await emptyGit.addConfig('user.name', 'John Doe');
+    await emptyGit.addConfig('user.email', 'john.doe@example.com');
+  });
+
+  afterAll(async () => {
+    await rm(baseDir, { recursive: true, force: true });
+  });
+
+  describe('without a branch and commits', () => {
+    it('getHashes should throw', async () => {
```

Comment:
Once it is clear this is the only function tested, its name does not need to be included in all the tests.

This should be modified in all other test cases as well.
```suggestion
    it('should throw', async () => {
```